autosklearn/data/validation.py

```diff
@@ -26,603 +21,115 @@ class InputValidator:
-            else:
-                self.target_encoder = make_column_transformer(
-                    (preprocessing.OrdinalEncoder(), list(range(y.shape[1]))),
-                )
-
-            # Mypy redefinition
-            assert self.target_encoder is not None
-
-            # remove ravel warning from pandas Series
-            if len(y.shape) > 1 and y.shape[1] == 1 and hasattr(y, "to_numpy"):
-                self.target_encoder.fit(y.to_numpy().ravel())
-            else:
-                self.target_encoder.fit(y)
-
-        try:
-            # remove ravel warning from pandas Series
-            if len(y.shape) > 1 and y.shape[1] == 1 and hasattr(y, "to_numpy"):
-                y = self.target_encoder.transform(y.to_numpy().ravel())
-            else:
-                y = self.target_encoder.transform(y)
-        except ValueError as e:
-            if 'Found unknown categories' in e.args[0]:
-                # Make the message more informative for Ordinal
-                raise ValueError(
-                    "During fit, the target array contained the categorical values {} "
-                    "which were encoded by Auto-sklearn automatically. "
-                    "Nevertheless, a new target set contained new categories not seen during "
-                    "training = {}. The OrdinalEncoder used by Auto-sklearn cannot handle "
-                    "this yet (due to a limitation on scikit-learn being addressed via:"
-                    " https://github.com/scikit-learn/scikit-learn/issues/17123)"
-                    "".format(
-                        self.target_encoder.transformers_[0][1].categories_,
-                        e.args[0],
-                    )
-                )
-            elif 'contains previously unseen labels' in e.args[0]:
-                # Make the message more informative
-                raise ValueError(
-                    "During fit, the target array contained the categorical values {} "
-                    "which were encoded by Auto-sklearn automatically. "
-                    "Nevertheless, a new target set contained new categories not seen during "
-                    "training = {}. This is a limitation in scikit-learn encoders being "
-                    "discussed in //github.com/scikit-learn/scikit-learn/issues/17123".format(
-                        self.target_encoder.classes_,
-                        e.args[0],
-                    )
-                )
-            else:
-                raise e
-
-        return y
-
-    def encode_target(
-        self,
-        y: np.ndarray,
-    ) -> np.ndarray:
-        """
-        Encodes the target if there is any encoder
-        """
-        if self.target_encoder is None:
-            return y
-        else:
-
-            return self.target_encoder.transform(y)
-
-    def decode_target(
+    ) -> BaseEstimator:
+        """
+        Validates and fit a categorical encoder (if needed) to the features, and
```

Comment:
Could you please extend the description to cover when exactly what is done?

Reply:
Done! please let me know if this should be extended more

Reply:
This looks good, there's no need to extend this further.